torch/distributed/nn/jit/instantiator.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python3
+import importlib
+import inspect
+import os
+
+import torch
+try:
+    # For fb.
+    from src.ATen.code_template import CodeTemplate
+except ImportError:
+    # For open-source.
+    from tools.shared.module_loader import import_module
+    CodeTemplate = import_module('code_template', 'aten/src/ATen/code_template.py').CodeTemplate
+from torch.distributed.nn.jit.templates import dir_path as TEMPLATE_DIR_PATH
+
+
+def get_return_type_from_callable(callable_obj):
+    sig = inspect.signature(callable_obj)
+    return_annotation = sig.return_annotation
+    if return_annotation is inspect.Signature.empty:
+        return callable_obj
+    return return_annotation
```

Comment:
I think we should have the following here:

1) Check if `callable_obj` is a class using `inspect.isclass()`. If this is true, we can just return callable_obj since this is the type of the Module that we're creating.
2) Check if callable_obj has return_annotation and if it does, return that as the type.
3) If 1) and 2) is not true, throw an exception mentioning that we couldn't determine the type of the Module and ask the user to add a return annotation.

Reply:
Updating.